rmgpy/thermo/nasa.pyx

```diff
@@ -361,8 +362,8 @@ cdef class NASA(HeatCapacityModel):
         # high temperature
         coeffs = numpy.zeros(15)
         coeffs[0] = polys[0].Tmax.value_si # mid point temperature between two polynomials
-        coeffs[1:8] = polys[1].coeffs # 7 coefficients of the high temperature polynomial
-        coeffs[8:15] = polys[0].coeffs # 7 coefficients of the low temperature polynomial
+        coeffs[1:8] = copy.deepcopy(polys[1].coeffs) # 7 coefficients of the high temperature polynomial
+        coeffs[8:15] = copy.deepcopy(polys[0].coeffs) # 7 coefficients of the low temperature polynomial
```

Comment:
Because you're putting into a slice of the coeffs array, rather than changing the object that the coeffs name points to, I'm pretty sure this is not necessary.
```python
In [1]: import numpy as np
In [2]: a=np.ones(5)
In [3]: b=np.zeros(3)
In [4]: a
Out[4]: array([ 1.,  1.,  1.,  1.,  1.])
In [5]: a[1:4]=b
In [6]: a
Out[6]: array([ 1.,  0.,  0.,  0.,  1.])
In [7]: b[1]=2
In [8]: a
Out[8]: array([ 1.,  0.,  0.,  0.,  1.])
In [9]: b
Out[9]: array([ 0.,  2.,  0.])
```
but it might not be expensive and I'm not sure how often this gets called.